tests/unit/services/cookies-test.js

```diff
@@ -87,6 +87,23 @@ describe('CookiesService', function() {
         expect(afterRoundtrip).to.eq(value);
       });
 
+      it('URI-component-decodes the value', function() {
+        let value = '!"§$%&/()=?"';
+        document.cookie = `${COOKIE_NAME}=${encodeURIComponent(value)}`;
+        let afterRoundtrip = this.subject().read(COOKIE_NAME);
+
+        expect(afterRoundtrip).to.eq(value);
+      });
+
+      it("doesn't decode the value when useRawValue is true", function() {
+        let value = '!"§$%&/()=?"';
+        let encodedValue = encodeURIComponent(value);
+        document.cookie = `${COOKIE_NAME}=${encodedValue}`;
```

Comment:
I guess it would be easier here to simple use the un-encoded value `document.cookie = `${COOKIE_NAME}=${value}`;` and change the assertion to `expect(afterRoundtrip).to.eq(value);`?

Reply:
Would that be a proper test for this scenario, though? Doesn't the set value need to be URI-encoded in order to verify that it's not URI-decoding it when `raw` is `true`?

Reply:
Yep, you're right of course 🤦‍♂️